apps/framework-docs/src/pages/moose/building/consumption-apis.mdx

```diff
@@ -78,15 +108,46 @@ class QueryParams(BaseModel):
 class ResponseBody(BaseModel):
     id: int
     name: str
+    value: float
+
+
+SourceTable = SourcePipeline.get_table()
 
 # Define the route handler function
-def run(params: QueryParams) -> ResponseBody:
-    ## Your logic here...
-    return ResponseBody(id=1, name="Example")
+def run(client: MooseClient, params: QueryParams) -> list[ResponseBody]:
+    query = f"""
+    SELECT 
+      id
+      name,
+      value
+    FROM {SourceTable.name}
+    WHERE category = {params.filterField}
+    LIMIT {params.maxResults}
+    """
+
+    return client.query.execute(query, {"filterField": params.filterField, "maxResults": params.maxResults})
```

Comment:
https://clickhouse.com/docs/integrations/python#common-method-arguments

it you value into the `f` string, you don't need it in the `parameters`